src/Microsoft.AspNetCore.Mvc.ViewFeatures/Controller.cs

```diff
@@ -216,6 +216,32 @@ public virtual PartialViewResult PartialView(string viewName, object model)
         /// The view component name. Can be a view component
         /// <see cref="ViewComponents.ViewComponentDescriptor.ShortName"/> or
         /// <see cref="ViewComponents.ViewComponentDescriptor.FullName"/>.</param>
+        /// <returns>The created <see cref="ViewComponentResult"/> object for the response.</returns>
+        [NonAction]
+        public virtual ViewComponentResult ViewComponent(string componentName)
```

Comment:
Think we have method overloads for a slew of other Action methods too. Let's stick with this pattern.